src/pages/trade/ui/route-book.tsx

```diff
@@ -1,60 +1,212 @@
-        <tbody>
-          {multiHops.sell.map(trace => (
-            <tr key={trace.price + trace.total} style={{ color: 'red' }}>
-              <td className='text-left tabular-nums'>{trace.price}</td>
-              <td className='text-right tabular-nums'>{trace.amount}</td>
-              <td className='text-right tabular-nums'>{trace.total}</td>
-              <td className='text-right tabular-nums'>{trace.hops.length}</td>
-            </tr>
-          ))}
-          {multiHops.buy.map(trace => (
-            <tr key={trace.price + trace.total} style={{ color: 'green' }}>
-              <td className='text-left tabular-nums'>{trace.price}</td>
-              <td className='text-right tabular-nums'>{trace.amount}</td>
-              <td className='text-right tabular-nums'>{trace.total}</td>
-              <td className='text-right tabular-nums'>{trace.hops.length}</td>
+    <div className='flex flex-col max-w-full border-y border-[#262626]'>
+      <div className='flex items-center gap-2 px-4 h-11 border-b border-[#262626]'>
+        <Tabs
+          value={'routes'}
+          onChange={() => {
+            undefined;
+          }}
+          options={ROUTEBOOK_TABS}
+          actionType='accent'
+        />
+      </div>
+
+      <div className='flex-1'>
+        <table className='w-full'>
+          <thead>
+            <tr className='text-xs text-gray-400'>
+              <th className='py-[8px] font-normal text-left'>Price({pair.quoteSymbol})</th>
+              <th className='py-[8px] font-normal text-right'>Amount({pair.baseSymbol})</th>
+              <th className='py-[8px] font-normal text-right'>Total</th>
+              <th className='py-[8px] font-normal text-right'>Route</th>
             </tr>
-          ))}
-        </tbody>
-      </table>
+          </thead>
+
+          <tbody className='relative'>
+            {multiHops.sell.map((trace, idx) => (
+              <TradeRow
+                key={`${trace.price}-${trace.total}-${idx}`}
+                trace={trace}
+                isSell={true}
+                relativeSize={sellRelativeSizes.get(trace.total) ?? 0}
+              />
+            ))}
+
+            <SpreadRow sellOrders={multiHops.sell} buyOrders={multiHops.buy} />
+
+            {multiHops.buy.map((trace, idx) => (
+              <TradeRow
+                key={`${trace.price}-${trace.total}-${idx}`}
+                trace={trace}
+                isSell={false}
+                relativeSize={buyRelativeSizes.get(trace.total) ?? 0}
+              />
+            ))}
+          </tbody>
+        </table>
+      </div>
     </div>
   );
 });
 
 export const RouteBook = observer(() => {
-  const { data, isLoading, error } = useBook();
+  const { data: bookData, isLoading: bookIsLoading, error: bookErr } = useBook();
 
-  if (error) {
-    return <div>Error loading route book: ${String(error)}</div>;
+  if (bookIsLoading || !bookData) {
+    return <div className='text-gray-400'>Loading...</div>;
```

Comment:
issue: if there is an error, the `!bookData` will guarantee it will be stuck in this loading state. Think the previous ordering prevented that possibility.

Reply:
question: Perhaps the loading skeleton can be included too? Or if not in this PR, perhaps a subsequent PR?